apps/st2-rules/controller.js

```diff
@@ -20,10 +20,10 @@ angular.module('main')
         url: '/new'
       })
       .state('rules.general', {
-        url: '/{ref:\\w+}/general?edit'
+        url: '/{ref:[\\w.-]+}/general?edit'
```

Comment:
Square brackets ([character class](http://www.regular-expressions.info/charclass.html)) in regexp implies literal characters.